components/Form/index.js

```diff
@@ -0,0 +1,332 @@
+import styled from "styled-components";
+import { css } from "styled-components";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+export default function Form({ onSubmit }) {
+  const [sleep, setSleep] = useState(true);
+  const [hygiene, setHygiene] = useState(false);
+  const [medicine, setMedicine] = useState(false);
+
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    for (const property in data) {
+      if (data[property].trim() === "") {
+        return alert("Bitte gib einen Text in das Feld ein!");
+      }
+    }
+
+    onSubmit(
+      data.name,
+      data.category,
+      data.adress,
+      data.link,
+      data.target,
+      data.requirements,
+      data.filterCriteria,
+      data.tel
+    );
+    alert("Die Einrichtung wurde erfolgreich hinzugefügt!");
+    router.push("/");
+  }
+
+  return (
+    <>
+      <button onClick={() => router.push("/")}>
+        {" "}
+        <svg
+          fill="#000000"
+          xmlns="http://www.w3.org/2000/svg"
+          viewBox="0 0 50 50"
+          width="40px"
+          height="40px"
+        >
+          <path d="M 9.15625 6.3125 L 6.3125 9.15625 L 22.15625 25 L 6.21875 40.96875 L 9.03125 43.78125 L 25 27.84375 L 40.9375 43.78125 L 43.78125 40.9375 L 27.84375 25 L 43.6875 9.15625 L 40.84375 6.3125 L 25 22.15625 Z" />
+        </svg>
+      </button>
+
+      <StyledForm onSubmit={handleSubmit}>
+        <fieldset>
+          <legend>Neue Einrichtungen hinzufügen:</legend>
```

Comment:
Please, have a look at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset), what a `fieldset` is used for:

> The `<fieldset>`  element is used to group several controls as well as labels `<label>` within a web form.

Think about, where in your form you really need `fieldsets` and do not use these just for styling reasons. 🙂 For example: The first `fieldset` wraps all content of your form. You do not need that one. I suggest to display the text "Neue Einrichtungen hinzufügen:" as a headline instead of a `legend`. If you like the look of that element, you can style the headline accordingly.

```jsx
<h2>Neue Einrichtungen hinzufügen:</h2>
   <StyledForm>
...
   </StyledForm>
```